assets/js/components/dashboard-sharing/ModuleRecoveryAlert/index.js

```diff
@@ -84,11 +111,13 @@ export default function ModuleRecoveryAlert() {
 		const modulesToRecover = Object.keys( checkboxes ).filter(
 			( module ) => checkboxes[ module ]
 		);
-		recoverModules( modulesToRecover ).finally( () => {
-			setRecoveringModules( false );
-			setCheckboxes( null );
+		clearErrors( 'recoverModule' ).then( () => {
+			recoverModules( modulesToRecover ).finally( () => {
+				setRecoveringModules( false );
+				setCheckboxes( null );
+			} );
 		} );
```

Comment:
Please use async/await for consistency. The `finally` shouldn't be necessary either as `recoverModules` will only throw if it's given invalid input. It won't throw if the request fails etc. `clearErrors` also won't throw. It isn't even async but all bound actions return promises as part of the wp data flow.

Reply:
Thank you for pointing out a best practice here! I'll keep it in mind for future cases. Updated it to use async/await.